location-service/src/main/java/com/movie/recommender/location/exception/handler/RestResponceEntityExceptionHandler.java

```diff
@@ -0,0 +1,32 @@
+package com.movie.recommender.location.exception.handler;
+
+import com.movie.recommender.location.exception.exceptions.LocationNotFoundException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.HttpStatus;
+import java.util.Map;
+
+@ControllerAdvice
+public class RestResponceEntityExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(DataIntegrityViolationException.class)
+    public ResponseEntity<?> handleDataIntegrityViolationException(DataIntegrityViolationException ex, WebRequest request) {
+        String message = "Duplicate entry for user_id. This user already has a location.";
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Map.of("status", "error", "message", message));
```

Comment:
update methods to use ErrorResponse dto
`return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ErrorResponse("error", message));`